cms/src/test/java/uk/nhs/digital/externalstorage/workflow/externalFilePublish/ExternalFilePublishTaskTest.java

```diff
@@ -30,33 +41,163 @@
+            .willReturn(true);
 
         Repository repository = MockJcr.newRepository();
         session = repository.login();
         rootNode = session.getRootNode();
+        testFolder = rootNode.addNode("lorem-ipsum", NT_FOLDER);
         externalFilePublishTask = new ExternalFilePublishTask();
 
         given(workflowContext.getInternalWorkflowSession()).willReturn(session);
         externalFilePublishTask.setWorkflowContext(workflowContext);
     }
 
     @Test
-    public void shouldCallS3() throws WorkflowException, RepositoryException {
-        Node docNode = rootNode.addNode("test-document");
-        addVariantNode(docNode, UNPUBLISHED);
-        MockJcr.setQueryResult(session, getQueryResults());
+    public void shouldPublishResources() throws WorkflowException, RepositoryException {
+        MockJcr.addQueryResultHandler(session, searchResourcesHandle(testExternalReference));
+
+        executeTask(getDocumentHandleWithVariant(getRandomDocumentType(), UNPUBLISHED), externalFilePublishTask);
+
+        then(s3Connector).should().publishResource(testExternalReference);
+        then(s3Connector).should(never()).unpublishResource(anyString());
+    }
+
+    @Test
+    public void shouldPublishFinalisedPublicationResources() throws WorkflowException, RepositoryException {
+        MockJcr.addQueryResultHandler(session, searchResourcesHandle(testExternalReference));
+
+        executeTask(getDocumentHandleWithVariant(getFinalisedPublication(), UNPUBLISHED), externalFilePublishTask);
+
+        then(s3Connector).should().publishResource(testExternalReference);
+        then(s3Connector).should(never()).unpublishResource(anyString());
+    }
+
+    @Test
+    public void shouldUnpublishUpcomingPublicationResources() throws WorkflowException, RepositoryException {
+        MockJcr.addQueryResultHandler(session, searchResourcesHandle(testExternalReference));
+
+        executeTask(getDocumentHandleWithVariant(getUpcomingPublication(), UNPUBLISHED), externalFilePublishTask);
+
+        then(s3Connector).should(never()).publishResource(anyString());
+        then(s3Connector).should().unpublishResource(testExternalReference);
+    }
+
+    @Test
+    public void shouldUnpublishDatasetResourcesPartOfUpcomingPublication() throws WorkflowException, RepositoryException {
+        MockJcr.addQueryResultHandler(session, searchResourcesHandle(testExternalReference));
+
+        // add Upcoming publication to JCR
+        getDocumentHandleWithVariant(getUpcomingPublication(), UNPUBLISHED);
+
+        executeTask(getDocumentHandleWithVariant(getDataset(), UNPUBLISHED), externalFilePublishTask);
+
+        then(s3Connector).should(never()).publishResource(anyString());
+        then(s3Connector).should().unpublishResource(testExternalReference);
+    }
+
+    @Test
+    public void shouldPublishDatasetResourcesPartOfFinalisedPublication() throws WorkflowException, RepositoryException {
```

Comment:
Isn't this test made redundant by `shouldUnpublishUpcomingPublicationAndItsDatasetsResources`?

Reply:
this is already done in `shouldUnpublishDatasetResourcesPartOfUpcomingPublication`